contents/tutorials/vue-surveys.md

```diff
@@ -0,0 +1,623 @@
+
+```vue file=components/CustomSurvey.vue
+<template>
+  <div class="survey">
+    <h2>{{ title }}</h2>
+    <div>
+      <button v-for="i in 10" :key="i" class="button" @click="handleSelect(i)">
+        {{ i }}
+      </button>
+    </div>
+    <div>
+      <button class="button" @click="emitDismiss">Dismiss</button>
+      <button class="button" @click="emitSubmit">Submit</button>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'CustomSurvey',
+  props: {
+    title: String,
+    onDismiss: Function,
+    onSubmit: Function
+  },
+  data() {
+    return {
+      selectedValue: null
+    };
+  },
+  methods: {
+    handleSelect(value) {
+      this.selectedValue = value;
+    },
+    emitDismiss() {
+      this.$emit('onDismiss');
+    },
+    emitSubmit() {
+      this.$emit('onSubmit', this.selectedValue);
+    }
+  }
+}
+</script>
+
+<style scoped>
+.survey {
+  position: fixed;
+  bottom: 20px;
+  right: 20px;
+  width: 400px;
+  padding: 20px;
+  background-color: #ffffff;
+  box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.2);
+  border-radius: 5px;
+  z-index: 1000;
+}
+
+.button {
+  margin: 5px;
+  padding: 5px;
+}
+</style>
+```
+
+Then, integrate the component into `App.vue`:
+
+```vue file=App.vue
+<template>
+  <main>
+    <div class="App">
+      <h1>This is our Vue.js survey tutorial</h1>
+      <CustomSurvey
+        v-if="showSurvey"
+        title="Survey title"
+        @onDismiss="handleDismiss"
+        @onSubmit="handleSubmit"
+      />
+    </div>
+  </main>
+</template>
+
+<script>
+import CustomSurvey from './components/CustomSurvey.vue'
+
+export default {
+  name: 'App',
+  components: {
+    CustomSurvey
+  },
+  data() {
+    return {
+      showSurvey: true
+    };
+  },
+  methods: {
+    handleDismiss() {
+      this.showSurvey = false;
+    },
+    handleSubmit(value) {
+      console.log("Submitted value:", value);
+      this.showSurvey = false;
+    }
+  }
+}
+</script>
+```
+
+This shows a survey popup every time you visit your app's homepage.
+
+![Custom survey UI](../images/tutorials/nextjs-surveys/sample-survey-ui.png)
+
+#### 2. Fetch the survey from PostHog
+
+PostHog keeps track of all active surveys for a user (this is especially helpful if you set up [custom targeting options](/docs/surveys/creating-surveys#targeting)). 
+
+To fetch the active surveys, we use `this.$posthog.getActiveMatchingSurveys()`:
+
+```vue file=App.vue
+<template>
+  <!-- ... rest of your template ... -->
+</template>
+
+<script>
+import CustomSurvey from './components/CustomSurvey.vue'
+
+export default {
+  name: 'App',
+  components: {
+    CustomSurvey
+  },
+  data() {
+    return {
+      showSurvey: true,
+    };
+  },
+  mounted() {
+    this.$posthog.getActiveMatchingSurveys((surveys) => {
+    });
+  },
+  methods: {
+    // ... rest of your methods ...
+  }
+}
+</script>
+
+<style>
+  /* ... rest of your styles ... */
+</style>
+
```

Comment:
We don't include styles in other parts of the tutorial so probably can remove it here.
```suggestion
```